alacritty_terminal/src/term/mod.rs

```diff
@@ -1063,15 +1065,21 @@ impl<T> Term<T> {
         &self.mode
     }
 
+    /// Swap primary and alternate screen buffer.
     pub fn swap_alt(&mut self) {
-        if self.alt {
+        if self.mode.contains(TermMode::ALT_SCREEN) {
             let template = self.grid.cursor.template;
             self.grid.region_mut(..).each(|c| c.reset(&template));
-        }
 
-        self.alt = !self.alt;
-        mem::swap(&mut self.grid, &mut self.alt_grid);
+            self.inactive_grid.cursor = self.inactive_grid.saved_cursor;
+            self.grid.cursor = self.grid.saved_cursor;
+        } else {
+            self.inactive_grid.saved_cursor = self.inactive_grid.cursor;
+            self.grid.saved_cursor = self.grid.cursor;
+        }
 
+        mem::swap(&mut self.grid, &mut self.inactive_grid);
+        self.mode ^= TermMode::ALT_SCREEN;
```

Comment:
I kind of liked that it was in `set` and `unset` methods, maybe we should keep it there? However, knowing that this function is being used only by those methods, maybe it's fine here.

Reply:
What did you like about it? Seems better to have it just in one place instead of duplicated.